src/overlays/actors/ovl_En_Holl/z_en_holl.c

```diff
@@ -24,37 +69,276 @@ const ActorInit En_Holl_InitVars = {
+            s32 playerSide = (this->actor.yDistToPlayer > 0.0f) ? EN_HOLL_ABOVE : EN_HOLL_BELOW;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/EnHoll_Init.s")
+            this->actor.room = globalCtx->doorCtx.transitionActorList[enHollId].sides[playerSide].room;
+            
+            if ((this->actor.room != globalCtx->roomCtx.currRoom.num) &&
+                Room_StartRoomTransition(globalCtx, &globalCtx->roomCtx, this->actor.room)) {
+                this->actionFunc = EnHoll_RoomTransitionIdle;
+                this->bgCoverAlphaActive = true;
+            }
+        }
+    } else if (this->bgCoverAlphaActive) {
+        this->bgCoverAlphaActive = false;
+        globalCtx->bgCoverAlpha = 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/EnHoll_Destroy.s")
+void EnHoll_VerticalIdle(EnHoll* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899ACC.s")
+    if (this->actor.xzDistToPlayer < EN_HOLL_RADIUS) {
+        f32 playerDistFromCentralPlane = fabsf(this->actor.yDistToPlayer);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899B88.s")
+        if (playerDistFromCentralPlane < EN_HOLL_ACTIVATION_PLANE_DISTANCE_VERTICAL &&
+            playerDistFromCentralPlane > EN_HOLL_LOADING_PLANE_DISTANCE_VERTICAL) {
+            s32 enHollId = EN_HOLL_GET_ID_CAST(this);
+            s32 playerSide = (this->actor.yDistToPlayer > 0.0f) ? EN_HOLL_ABOVE : EN_HOLL_BELOW;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_80899F30.s")
+            this->actor.room = globalCtx->doorCtx.transitionActorList[enHollId].sides[playerSide].room;
+            if ((this->actor.room != globalCtx->roomCtx.currRoom.num) &&
+                Room_StartRoomTransition(globalCtx, &globalCtx->roomCtx, this->actor.room)) {
+                this->actionFunc = EnHoll_RoomTransitionIdle;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_8089A0C0.s")
+void EnHoll_RoomTransitionIdle(EnHoll* this, GlobalContext* globalCtx) {
+    if (globalCtx->roomCtx.unk31 == 0) {
+        func_8012EBF8(globalCtx, &globalCtx->roomCtx);
+        if (globalCtx->bgCoverAlpha == 0) {
+            this->bgCoverAlphaActive = false;
+        }
+        EnHoll_SetupAction(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_8089A238.s")
+void EnHoll_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnHoll* this = THIS;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/func_8089A330.s")
+    if ((globalCtx->sceneLoadFlag == 0) && (globalCtx->unk_18B4A == 0) && !(player->stateFlags1 & 0x200)) {
+        this->actionFunc(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Holl/EnHoll_Update.s")
+void EnHoll_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnHoll* this = THIS;
+    Gfx* dl;
+    u32 dlIndex;
```

Comment:
I think my personal preference would be to use either `Gfx* gfx` or `Gfx* dList` as temps instead of `Gfx* dl` since I'm always cautious of over-abbreviation. But `dl` is fine as is for now. Just putting my thoughts out there for future discussions

Reply:
I had originally named it gfxP at somebody's suggestion, but then in code review they told me that gfxP is the name for an array of Gfx*, so Angehlo suggested I call it "`dlist` or something similar" (see: https://github.com/zeldaret/mm/pull/270#discussion_r689011853), so I took `dl` to be similar, and thought it matched better with `dlIndex`. I can switch to dList though if you prefer.

Reply:
Alternatively, if I don't change it... is the capitalization correct, or would `dL` and `dLIndex` be the proper camelCase?